### ISSUE TITLE
Don't flag conditional constant definitions (`reportConstantRedefinition`/`reportRedeclaration`)

### ISSUE THREAD

**Issue by RBerga06** [DetachHead/basedpyright#1657]
### Description

Apparently basedpyright does not support conditional constants (nor conditional class / type alias definitions). For example:

```python
# pyright: reportUnreachable=false
import sys

class MacFoo: pass
class WinFoo: pass
class LinFoo: pass


if sys.platform == "darwin":
    OS = "macOS"
    type SysFoo = MacFoo  # reportRedeclaration
    class SysBar: pass    # reportRedeclaration

elif sys.platform == "win32":
    OS = "Windows"        # reportConstantRedefinition
    type SysFoo = WinFoo  # reportRedeclaration
    class SysBar: pass    # reportRedeclaration

else:
    OS = "Linux"          # reportConstantRedefinition
    type SysFoo = LinFoo
    class SysBar: pass
```
([`basedpyright` playground](https://basedpyright.com/?typeCheckingMode=all&code=MQAgDgngTglg5gCwC4C4RQKZgPZSQVQDtMBDAYwRICMAbDAXgDMSaBnDAKBgFsc8RWEVhw5kaJVqxABZcgDFs2NGAnCxqkAHUYhBUvCrR4ySAAyOvcsMiYjAUIB0YcUka5uIevRAAiACYkUADuOj4oHCCRIADyAMqevtzkcT4RUUgQYBggsUJ6CbJk%2BSCgmHxIAEoYfhjqUCRIMNiEaZHqJrmsAEKBViZRpVi4ldW14vWNzSIYNLb2rE4ublAeXr4hhADMAExhrTHx3j7ahH7YQaw%2BUdcl6EN4AMLNrEgkhCM1jDowky3XGVkcnlFAkTsVBuUqjU6g0mn8ou0pJ0elA%2BlIBndIaMYb9pmwMOFrnEEj5zIQAK4ADyuN2uEOGT0ILzeHwwX0IPzh%2BwB2U6%2BW8ZL0%2B0RQO6vQMkiAA))

This issue also affects pyright ([playground link](https://pyright-play.net/?strict=true&code=MQAgDgngTglg5gCwC4C4RQKZgPZSQVQDtMBDAYwRICMAbDAXgDMSaBnDAKBgFsc8RWEVhw5kaJVqxABZcgDFs2NGAnCxqkAHUYhBUvCrR4ySAAyOvcsMiYjAUIB0YcUka5uIevRAAiACYkUADuOj4oHCCRIADyAMqevtzkcT4RUUgQYBggsUJ6CbJk%2BSCgmHxIAEoYfhjqUCRIMNiEaZHqJrmsAEKBViZRpVi4ldW14vWNzSIYNLb2rE4ublAeXr4hhADMAExhrTHx3j7ahH7YQaw%2BUdcl6EN4AMLNrEgkhCM1jDowky3XGVkcnlFAkTsVBuUqjU6g0mn8ou0pJ0elA%2BlIBndIaMYb9pmwMOFrnEEj5zIQAK4ADyuN2uEOGT0ILzeHwwX0IPzh%2BwB2U6%2BW8ZL0%2B0RQO6vQMkiAA)), but this was considered expected behaviour (upstream issue: https://github.com/microsoft/pyright/issues/5265).

**Comment by DetachHead** [comment]
hmm i thought we had an issue for this already but i guess not. thanks
